pandas/_libs/hashtable.pyi

```diff
@@ -231,15 +232,15 @@ class IntpHashTable(HashTable): ...
 def duplicated(
     values: np.ndarray,
     keep: Literal["last", "first", False] = ...,
-    mask: npt.NDArray[np.bool_] | None = ...,
+    mask: npt.NDArray[np.bool_] | BitmaskArray | None = ...,
```

Comment:
For typing annotations like this we could also do a precursor to change `mask` and other arguments that ared typed memoryviews in Cython to be `typing_extensions.Buffer`; I think that would be more technically correct

Python 3.12 will have collections.abc.Buffer built in. See also PEP 688 https://peps.python.org/pep-0688/#equivalent-for-older-python-versions

I think the only loss that way is I don't see the typing.Buffer as being Generic, so AFAICT can't do typing.Buffer[bool] yet. But seems logical that will get implemented in Python at some point